ain/ain.py

```diff
@@ -64,39 +64,157 @@ def setSigner(self, signer: Signer):
 
     # TODO(kriii): Return objects typing.
 
-    async def getBlock(
+    async def getLastBlock(self) -> Any:
+        """Fetches the last block.
+
+        Args:
+
+        Returns:
+            The last block.
+        """
+        return await self.provider.send(
+            "ain_getLastBlock", {}
+        )
+
+    async def getLastBlockNumber(self) -> int:
+        """Fetches the last block number.
+
+        Args:
+
+        Returns:
+            The last block number.
+        """
+        return await self.provider.send(
+            "ain_getLastBlockNumber", {}
+        )
+
+    async def getBlockByNumber(
         self,
-        blockHashOrBlockNumber: Union[str, int],
+        blockNumber: int,
         returnTransactionObjects: bool = False,
     ) -> Any:
-        """Gets a block with the given hash or block number.
+        """Gets a block with the given block number.
 
         Args:
-            blockHashOrBlockNumber (Union[str, int]): The block hash or the block number.
+            blockNumber (int): The block number.
             returnTransactionObjects (bool): If `True`, returns the full transaction objects.
                 If `False`, returns only the transaction hashes. Default to `False`.
         
         Returns:
-            The block with the given hash or block number.
+            The block with the given block number.
         """
-        if type(blockHashOrBlockNumber) is str:
-            return await self.provider.send(
-                "ain_getBlockByHash",
-                {
-                    "getFullTransactions": returnTransactionObjects,
-                    "hash": blockHashOrBlockNumber,
-                },
-            )
-        elif type(blockHashOrBlockNumber) is int:
-            return await self.provider.send(
-                "ain_getBlockByNumber",
-                {
-                    "getFullTransactions": returnTransactionObjects,
-                    "number": blockHashOrBlockNumber,
-                },
-            )
-        else:
-            raise TypeError("blockHashOrBlockNumber has invalid type")
+        return await self.provider.send(
+            "ain_getBlockByNumber",
+            {
+                "getFullTransactions": returnTransactionObjects,
+                "number": blockNumber,
+            },
+        )
+
+    async def getBlockByHash(
+        self,
+        blockHash: str,
+        returnTransactionObjects: bool = False,
+    ) -> Any:
+        """Gets a block with the given block hash.
+
+        Args:
+            blockHash (str]): The block hash.
```

Comment:
may it's typo

Reply:
right, it'll be fixed in the coming pr. thanks!